staging/src/k8s.io/apimachinery/pkg/api/apitesting/roundtrip/construct.go

```diff
@@ -173,6 +173,10 @@ func fill(dataString string, dataInt int, t reflect.Type, v reflect.Value, fillF
 		// use Convert to set into int alias types and different int widths correctly
 		v.Set(reflect.ValueOf(dataInt).Convert(t))
 
+	case reflect.Float32, reflect.Float64:
+		// use Convert to set into float types
+		v.Set(reflect.ValueOf(dataInt).Convert(t))
```

Comment:
make sure this exercises the float round tripping with something like this:

```suggestion
		v.Set(reflect.ValueOf(float32(dataInt) + 0.5).Convert(t))
```

Reply:
updated code as sugguesed. thanks.